src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/UpToDate/BuildUpToDateCheck.cs

```diff
@@ -119,11 +119,7 @@ private void OnProjectChanged(IProjectSubscriptionUpdate e)
             _msBuildProjectFullPath = e.CurrentState.GetPropertyOrDefault(ConfigurationGeneral.SchemaName, ConfigurationGeneral.MSBuildProjectFullPathProperty, _msBuildProjectFullPath);
             _msBuildProjectDirectory = e.CurrentState.GetPropertyOrDefault(ConfigurationGeneral.SchemaName, ConfigurationGeneral.MSBuildProjectDirectoryProperty, _msBuildProjectDirectory);
             _outputRelativeOrFullPath = e.CurrentState.GetPropertyOrDefault(ConfigurationGeneral.SchemaName, ConfigurationGeneral.OutputPathProperty, _outputRelativeOrFullPath);
-
-            string[] allProjects = e.CurrentState.GetPropertyOrDefault(ConfigurationGeneral.SchemaName, ConfigurationGeneral.MSBuildAllProjectsProperty, string.Empty)
-                .Split(new[] { ';' }, StringSplitOptions.RemoveEmptyEntries);
-            _imports.Clear();
-            _imports.AddRange(allProjects);
+            _msBuildAllProjects = e.CurrentState.GetPropertyOrDefault(ConfigurationGeneral.SchemaName, ConfigurationGeneral.MSBuildAllProjectsProperty, "");
```

Comment:
@davkean does this repo have a stated preference for `string.Empty`/`String.Empty`/`""`?

Reply:
Find in files shows:

- 94 instances of `string.Empty`
- 119 instances of `String.Empty`
- 2575 instances of `""` (probably lots of false positives)

I don't feel strongly either way, though tend to use `""`.

Reply:
String 🤮

Its the wrong colour! How can anyone live like this :P

Reply:
I vote `string.Empty` please

Reply:
Sure. I won't change any more of these.

Reply:
There are people who think that you should use `string` as a type name for variables, but `String` any time you access a static member off of the type.  I'm not sure *why* people think that, but some do.  🤷‍♂️